[PAY-1790] Create Backing With Setup Intent Client Secret

msadoon opened the pull request:
# 📲 What

This PR wraps up the major portion of work to use a newly created card from the Stripe Payment Sheet to complete a backing within the iOS app.

# 🤔 Why

Part of the ongoing Epic to include Stripe in our payments systems so we can offload much of the complexity of payment processing to an expert vendor.

# 🛠 How

Previously used `creditCardSelectedSignal` is updated to use a tuple that knows if the payment source id is coming from an existing payment source or new payment sheet source.

Leaving `selectedPaymentSourceId` as is to update the `willRetryPaymentMethod` having mapped `creditCardSelectedSignal` to its `paymentSourceId`

Also `creditCardSelectedSignal` is mapping to its `paymentSourceid` whereever its used except in the new signal `selectedPaymentSourceIdOrSetupIntentClientSecret` 

That signal is used directly in the creation of the `creatingBackingEvents` and `updateBackingEvents` network requests, the result of which is passed on to SCA flow detection or thank you page creation.

When `CreateBackingData` is created we take precedence of 1. Apple Pay, 2. Payment Source ID, 3. Setup Intent Client Secret. Only one of these values is passed to the `CreateBackingInput` mutation.

# 👀 See

Before 🐛 

https://user-images.githubusercontent.com/4282741/185483536-8d43c559-44ed-41c8-82b4-791a2c46d446.MP4

After 🦋

https://user-images.githubusercontent.com/4282741/185484795-f893ce5d-edb5-4328-8816-f42fab11c333.MP4


# ✅ Acceptance criteria

- [x] Ensure create backing works with newly created card from payment sheet.

# ⏰ TODO

- [x] Fix happy path (vanilla `CreateBacking` with `setupIntentClientSecret`)
- [x] Ensure other contexts besides `.pledge` never see the Payment Sheet
- [x] Ensure the 3DS flow and Apple Pay still work as expected (in code)
- [x] Ensure cards used to pledge are retained in both settings and in the manage pledge page. 
- [x] Ensure change payment methods, updating reward, updating pledge amount, (fix is kinda hard to test, it should work -- see if I can spoof it) are all working
- [x] Write unit tests

msadoon committed:
setup happy path for create backing. Not working for some reason with just setup intent client secret, needs investigation.

msadoon committed:
minor updates to GQL schema

msadoon committed:
fixed an issue wrt to confirming the payment method before allowing the pledge to be made. CreateBacking working.

msadoon committed:
wrote tests for business logic

nativeksr commented:
<!--
  0 Errors
  1 Warning: Big PR
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

msadoon committed:
added tests for other contexts besides pledge

Ensuring payment sheet only shows for the pledge context with the feature flag on.

msadoon committed:
fixed formatting
